src/cmd-buildextend-live

```diff
@@ -418,6 +418,7 @@ def generate_iso():
         kargs_json.update(
             size=karg_embed_area_length,
             default=cmdline.strip(),
+            arch=basearch,
```

Comment:
we can't find this information any other way than setting it here?

Reply:
in general we can check smth like `[[ -f images/pxeboot/kernel.img ]] && file images/pxeboot/kernel.img` to detect `s390x` arch (x86 uses `images/pxeboot/vmlinuz` )

Reply:
The reason we actually need this is because https://github.com/coreos/coreos-installer/pull/1230 needs to know the padding byte for the kargs array, right?  But encoding that as `s390x` requires coreos-installer to assume that non-s390x always wants `#` and s390x always wants `\0`, which isn't guaranteed to always be true.  Instead, can we add an optional field to the entries for the individual kargs areas, which specifies the padding byte?

Reply:
@bgilbert s390x doesn't use '\n', so only padding byte wouldn't be enough.

Reply:
Can we define some sort of keyword that describes the padding, then?

Reply:
Do you mean smth like :
```
  "arch": "ABC",
  "padding":acsii-code-number
  "default": "karg1 karg2",
  "files": [ ...  ],
  "size": XYZ
}

``` ?

Reply:
No, the idea is to drop the `arch` field, since that's not what we actually care about.  We should avoid hardcoding assumptions about what padding is needed by what architectures, since we might use different bootloaders in the future.

I meant something like this:

```json
{
  "default": "mitigations=auto,nosmt coreos.liveiso=fedora-coreos-38.20230625.3.0 ignition.firstboot ignition.platform.id=metal",
  "files": [
    {
      "offset": 970,
      "path": "EFI/fedora/grub.cfg",
      "pad": "hash-newline"
    },
    {
      "offset": 1869,
      "path": "isolinux/isolinux.cfg",
      "pad": "nul"
    }
  ],
  "size": 1137
}
```

As an alternative, we could have multiple attributes like `"pad": "#", "end": "\n"`, but that's both too general (we don't need to support arbitrary characters) and not general enough (maybe in the future there's another factor that isn't covered by those two fields).  So I think it could make sense to define names for the two behaviors we need right now.